suite-native/thp/src/hooks/useThpAutoconnectAlerts.tsx

```diff
@@ -0,0 +1,85 @@
+import { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { removeThpAutoconnectThunk, startThpAutoconnectThunk, thpActions } from '@suite-common/thp';
+import { selectSelectedDevice } from '@suite-common/wallet-core';
+import { useAlert } from '@suite-native/alerts';
+import { Translation } from '@suite-native/intl';
+import { useToast } from '@suite-native/toasts';
+
+export const useThpAutoconnectAlerts = () => {
+    const dispatch = useDispatch();
+    const { showAlert } = useAlert();
+    const { showToast } = useToast();
+
+    const device = useSelector(selectSelectedDevice);
+
+    const hasAutoconnectCredential =
+        device?.thp?.credentials.some(credential => credential.autoconnect) ?? false;
+
+    const turnOnAutoconnect = useCallback(() => {
+        dispatch(startThpAutoconnectThunk());
+    }, [dispatch]);
+
+    const turnOffAutoconnect = useCallback(async () => {
+        const result = await dispatch(
+            removeThpAutoconnectThunk({
+                credentials: device?.thp?.credentials.filter(cred => cred.autoconnect) ?? [],
+            }),
+        ).unwrap();
+
+        if (!result?.success) {
+            showToast({
+                variant: 'error',
+                message: result?.payload.error,
+            });
+        }
```

Comment:
This seems wrong. The `removeThpAutoconnectThunk` always returns fulfilled values. It should return `rejectWithValue` instead for better debugging and error handling. I understand this is not code you've added though, so feel free to skip if we're in a rush with this.